agent/xds/rbac.go

```diff
@@ -233,36 +273,43 @@ func intentionToIntermediateRBACForm(
 	}
 
 	if isHTTP && ixn.JWT != nil {
-		var c []*JWTInfo
+		var jwts []*JWTInfo
 		for _, prov := range ixn.JWT.Providers {
-			if len(prov.VerifyClaims) > 0 {
-				c = append(c, makeJWTInfos(prov, nil, 0))
+			jwtProvider, ok := pCE[prov.Name]
+
+			if !ok {
+				return nil, fmt.Errorf("provider specified in intention does not exist. Provider name: %s", prov.Name)
 			}
+			jwts = append(jwts, makeJWTInfos(prov, jwtProvider))
 		}
-		if len(c) > 0 {
-			rixn.jwtInfos = c
+
+		if len(jwts) > 0 {
```

Comment:
Was there a reason to check if that this length is greater than 0? I think we can remove the check and directly assign to `rixn.jwtInfos`.